processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -462,7 +462,7 @@ private Assignment assignToCollection(Type targetType,
                     // target accessor is setter, so wrap the setter in setter map/ collection handling
                     result = new SetterWrapperForCollectionsAndMaps(
                         result,
-                        Executables.getCollectionGetterName( targetWriteAccessor ),
+                        targetReadAccessor.getSimpleName().toString(),
```

Comment:
dunno.. Indeed apparently not 😉  But perhaps @gunnarmorling knows whether there was some kind of `deeper` reason why this ended up in the SPI. If that's not the case, we should not do it..